### PR TITLE
RoR capstone project

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [RoR capstone project](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md)

## Hi good reviewer, I know I added a few integration tests but I am running out of time. I will keep adding them please help me. 🙂

Link to [Video](https://www.loom.com/share/86366f22126d4136ad32036b45bae0de)

### Setup the project to review it:

These commands will allow you to run the app in your computer:
2. Clone the app in your computer (only one branch) with the following commands.
```
  git clone -b feature/project-details --single-branch git@github.com:MarcoOquendoC/Budget-app.git
  cd Budget-app
  bundle install
  npm install
```
3. Type your **PostgreSQL password** in config/database.yml (line 6).
```
  rails db:create
  rails db:migrate
  rails db:seed
  rails tailwindcss:build
  rails db:migrate RAILS_ENV=test
  rails db:seed RAILS_ENV=test
  rspec
  rails s
```

P and Bob thinking

## This Pull Request proposes the implementation described [in this requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) by Microverse.

I changed the splash name to home to add it in the future to the navbar and also an About page. I hope it does not make problems since the requirement said "splash".

Requirements:

*Splash screen (I called it home)*
- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
 
*Sign up and log in pages*
- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

*Home page (catagories page)*
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

*Transactions page*
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

*"Add a new category" page*
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

*"Add a new transaction" page*
- The user fills out a form to create a new transaction with:
  - name (mandatory)
  - amount (mandatory)
  - categories (mandatory at least one)
- The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

**Review by 1GORDON — COMMENTED** [review]
##
## Hi @MarcoOquendoC ✋

Good job so far!

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# This project will be marked as invalid

- [x] Kindly add the correct [readme template](https://github.com/microverseinc/readme-template) and customize it accordingly.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarcoOquendoC** [comment]
# Hi @1GORDON.

Thank you for the review. I believe you where talking about my PR message and I have fixed it.

File: spec/requests/categories_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                     icon: 'https://www.flaticon.com/svg/static/icons/svg/3144/3144456.svg' },
+                   { author: @user, name: 'Appliances', icon: 'https://www.flaticon.com/svg/static/icons/svg/3144/3144456.svg' }
+                 ])
+
+    sign_in @user
+    get groups_path
+  end
+
+  it 'assigns all categories to @groups' do
+    expect(assigns(:groups)).to eq(@user.groups)
```
Comment (BenMukebo) [inline spec/requests/categories_spec.rb:18]: - [x] Excellent addition of unit tests for all the most important components of the RoR application. 👍🏻 However, I get error messages even after following the instructions written in the README file to run the test. Please correct them 👍🏻

image

- [ ] Now, let's make sure to also add integration tests with capybara for all the most important components of the RoR application. ✍🏻

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @MarcoOquendoC, 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🥇 
- Git flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Great job on the validations. ✔️ 
- Correctly use Devise for authentication. ✔️ 
- Your app has been well deployed on render . ✔️ 
- Your Design looks awesome, keep rocking :rocket: 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/categories_spec.rb (hunk 1 above)
Comment (MarcoOquendoC) [reply spec/requests/categories_spec.rb:18]: Hi @BenMukebo,

I review the steps I sent and it works properly.

I added a few integration tests but I am running out of time. I will keep working on it.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @MarcoOquendoC ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**